docs/question/template.md

```diff
@@ -1,6 +1,34 @@
-# Using Markdown in questions
+# `question.html`
 
-HTML and custom elements are great for flexibility and expressiveness. However, they're not great for working with large amounts of text, formatting text, and so on. [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet) is a lightweight plaintext markup syntax that's ideal for authoring simple but rich text. To enable this, PrairieLearn adds a special `<markdown>` tag to questions. When a `<markdown>` block is encountered, its contents are converted to HTML. Here's an example `question.html` that utilizes this element:
+## Template file
+
+All `question.html` files are processed with the [Mustache template engine](https://mustache.github.io/mustache.5.html), which allows for basic dynamic rendering of HTML. For example, you can use the following syntax to conditionally render a piece of HTML:
+
+```html title="question.html"
+{{#params.show}}
+<p>This will only show up if params.show is true</p>
+{{/params.show}}
+```
+
+and this syntax to render over an array of strings (the `.` represents the current item in the array):
+
+```html title="question.html"
+{{#params.items}}
+<p>{{.}}</p>
+{{/params.items}}
+```
+
+!!! tip
+
+     If you want to render student-provided HTML, we recommend the use of the [`<pl-xss-safe>` element](../elements.md#pl-xss-safe-element).
+
+     You can also use triple-braces (e.g. `{{{params.html}}}`) to substitute raw HTML. This is not recommended, as it can lead to XSS vulnerabilities.
```

Comment:
I think saying that it is not recommended may be too strong (note that it has the same "security" issues as the template itself if used with instructor-provided HTML). Maybe point out that this should only be used for content that is defined by the instructor, and not for student-provided content?

Also, I believe the pl-xss-safe comment should come after this point, as an alternative for student-provided content.

Reply:
Did this in d593c196707fc85bfc70bbde053f00454c582183. I also broke the note out into a section, since it's now a larger note.